.travis.yml

```diff
@@ -0,0 +1,164 @@
+# Copy these contents into the root directory of your Github project in a file
+# named .travis.yml
+
+# Use new container infrastructure to enable caching
+sudo: false
+
+compiler:
+  gcc
+
+# Enable HLint for haskell at CodeClimate
+engines:
+  hlint:
+    enabled: true
+ratings:
+  paths:
+  - "**.hs"
+
+# Choose a lightweight base image; we provide our own build tools.
+language: c
+
+addons:
+  apt:
+    packages:
+    - libgmp-dev
+    - libmpfr-dev
+    - libmpc-dev
+    - binutils-dev
+    - g++-4.7
+    - gcc
+
+# Caching so the next build will be fast too.
+cache:
+  directories:
+  - $HOME/.ghc
+  - $HOME/.cabal
+  - $HOME/.stack
+
+# The different configurations we want to test. We have BUILD=cabal which uses
+# cabal-install, and BUILD=stack which uses Stack. More documentation on each
+# of those below.
+#
+# We set the compiler values here to tell Travis to use a different
+# cache file per set of arguments.
+#
+# If you need to have different apt packages for each combination in the
+# matrix, you can use a line such as:
+#     addons: {apt: {packages: [libfcgi-dev,libgmp-dev]}}
+matrix:
+  include:
+  # We grab the appropriate GHC and cabal-install versions from hvr's PPA. See:
+  # https://github.com/hvr/multi-ghc-travis
+  - env: BUILD=cabal GHCVER=7.8.4 CABALVER=1.18 CC=gcc-4.7 CXX=g++-4.7
+    compiler: ": #GHC 7.8.4"
+    addons: {apt: {packages: [cabal-install-1.18,ghc-7.8.4], sources: [hvr-ghc]}}
```

Comment:
When you are defining apt addons here, global ones are overriden.
You have to copy all of those here or use yaml references. https://github.com/travis-ci/travis-ci/issues/3505#issuecomment-92764855